addons/ekco/template/base/install.sh

```diff
@@ -230,64 +230,101 @@ function ekco_handle_load_balancer_address_change_post_init() {
 # temporarily run haproxy as a container directly with docker or containerd.
 function ekco_bootstrap_internal_lb() {
     local backends="$PRIMARY_HOST"
-    if [ -z "$backends" ]; then
+    if [ -z "$backends" ] && [ "$MASTER" = "1" ]; then
         backends="$PRIVATE_ADDRESS"
     fi
 
+    # Check if load balancer is already bootstrapped before we update the manifests
+    # which will cause the load balancer to restart.
+    local already_bootstrapped=0
+    local last_modified=
+    if curl -skf https://localhost:6444/healthz >/dev/null ; then
+        already_bootstrapped=1
+        last_modified="$(stat -c %Y /etc/kubernetes/manifests/haproxy.yaml)"
+    fi
+
+    echo "Updating the Kubernetes apiserver load balancer"
+
     # Always regenerate the manifests to account for updates.
     # Note: this could cause downtime when kublet syncs the manifests for a new haproxy version
     if commandExists docker; then
         mkdir -p /etc/kubernetes/manifests
         docker run --rm \
             --entrypoint="/usr/bin/ekco" \
             --volume '/etc:/host/etc' \
-            replicated/ekco:v$EKCO_VERSION \
+            "replicated/ekco:v$EKCO_VERSION" \
             generate-haproxy-manifest \
                 --file=/host/etc/kubernetes/manifests/haproxy.yaml \
-                --image="${EKCO_HAPROXY_IMAGE}"
+                --image="$EKCO_HAPROXY_IMAGE"
     else
         mkdir -p /etc/kubernetes/manifests
-        ctr --namespace k8s.io run --rm \
+        if [ "$AIRGAP" != "1" ]; then
+            # the image will not be loaded from the add-on directory and thus will not exist in the dev environment
+            ctr -n k8s.io images pull "docker.io/replicated/ekco:v$EKCO_VERSION" >/dev/null
+        fi
```

Comment:
i will check for the existence of that image. the issue is without this the code will not work in the dev environment because of the image archives do not exist

```
⚙  Addon containerd 1.6.8
find: ‘./addons/weave/2.6.5/images’: No such file or directory
find: ‘./addons/ekco/0.23.1/images’: No such file or directory
⚙  Addon ekco 0.23.1
```